ydb/core/blobstorage/pdisk/blobstorage_pdisk_tools.cpp

```diff
@@ -233,10 +233,19 @@ bool ReadPDiskFormatInfo(const TString &path, const NPDisk::TMainKey &mainKey, T
 }
 
 void ObliterateDisk(TString path) {
-    TFile f(path, OpenAlways | RdWr);
+    TFile f(path, OpenExisting | RdWr);
     f.Flock(LOCK_EX | LOCK_NB);
 
-    TVector<ui8> zeros(NPDisk::FormatSectorSize * NPDisk::ReplicationFactor, 0);
+    bool isBlockDevice = false;
+    ui64 diskSizeBytes = 0;
+    DetectFileParameters(path, diskSizeBytes, isBlockDevice);
+
+    constexpr size_t portionSize = NPDisk::FormatSectorSize * NPDisk::ReplicationFactor;
+    if (diskSizeBytes <= portionSize) {
+        ythrow TFileError() << "illegal size " << diskSizeBytes << " for device " << path;
```

Comment:
The PR description seems to me more clear than words "illegal size", I suggest citing it as is. Also, `yexception` seems to be more appropriate here.  

```suggestion
        ythrow yexception() << "file is too small to be the YDB storage device, path# " << path.Quote() <<
            " diskSizeBytes# " << diskSizeBytes;
```